Sources/PopOverViewController.swift

```diff
@@ -72,51 +127,123 @@ open class PopOverViewController: UITableViewController, UIAdaptivePresentationC
         return titles.count
     }
     
+    override open func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        return heightPerItem
+    }
     
     override open func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
         var cell: UITableViewCell
         
         let title:String? = titles[indexPath.row]
         
         // If explanation text is coming, display it in two lines
-        if (descriptions == nil) {
-            cell = tableView.dequeueReusableCell(withIdentifier: "SingleTitleCell")!
-            cell.textLabel?.text = title
+        var reuseIdentifier: String
+        var description:String?
+        var cellStyle: UITableViewCellStyle = .default
+        if let descriptions = descriptions, descriptions.count > indexPath.row, !descriptions[indexPath.row].isEmpty {
+            reuseIdentifier = "SubTitleCell"
+            description = descriptions[indexPath.row]
+            cellStyle = .subtitle
         } else {
-            let description:String? = descriptions?[indexPath.row]
+            reuseIdentifier = "SingleTitleCell"
+        }
+        if let c = tableView.dequeueReusableCell(withIdentifier: reuseIdentifier) {
+            cell = c
+        } else {
+            cell = UITableViewCell(style: cellStyle, reuseIdentifier: reuseIdentifier)
+        }
+
+        cell.textLabel?.text = title
+        if let text = description {
+            cell.detailTextLabel?.text = text
+        }
```

Comment:
Why is this only assigned when description is not null ?
Shouldn't this be cell.detailTextLabel?.text = description ?? ""
If this stays like this, and the cell was dequeued using reuse identifier, chances are it will still show the text from the previous object it was linked with

Reply:
That's true